pages/docs/reference/java-interop.md

```diff
@@ -131,6 +131,135 @@ Kotlin types. The compiler supports several flavors of nullability annotations,
+`When.NEVER` denote a nullable type; and `When.UNKNOWN` forces the type to be [platform one](#null-safety-and-platform-types).
+
+A library can be compiled against the JSR-305 annotations, but there's no need to make the annotations artifact (e.g. `jsr305.jar`)
+a compile dependency for the library consumers. The Kotlin compiler can read the JSR-305 annotations from a library without the annotations 
+present on the classpath.
+
+Since Kotlin 1.1.60, 
+[custom nullability qualifiers (KEEP-79)](https://github.com/Kotlin/KEEP/blob/41091f1cc7045142181d8c89645059f4a15cc91a/proposals/jsr-305-custom-nullability-qualifiers.md) 
+are also supported (see below).
+
+##### Type qualifier nicknames (since Kotlin 1.1.60)
+
+If an annotation type is annotated with both
+[`@TypeQualifierNickname`](https://aalmiray.github.io/jsr-305/apidocs/javax/annotation/meta/TypeQualifierNickname.html) 
+and JSR-305 `@Nonnull` (or its another nickname, such as `@CheckForNull`), then the annotation type is itself used for 
+retrieving precise nullability and has the same meaning as that nullability annotation:
+
+``` java
+@TypeQualifierNickname
+@Nonnull(when = When.ALWAYS)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface MyNonnull {
+}
+
+@TypeQualifierNickname
+@CheckForNull // a nickname to another type qualifier nickname
+@Retention(RetentionPolicy.RUNTIME)
+public @interface MyNullable {
+}
+
+interface A {
+    @MyNullable
+    String foo(@MyNonnull String x);
+}
+```
+
+`A::foo` in this example is available in Kotlin as `fun foo(x: String): String?`
+
+##### Type qualifier defaults (since Kotlin 1.1.60)
+
+[`@TypeQualifierDefault`](https://aalmiray.github.io/jsr-305/apidocs/javax/annotation/meta/TypeQualifierDefault.html) 
+allows introducing annotations that, when being applied, define the default nullability within the scope of the annotated 
+element.
+
+Such annotation type should itself be annotated with both `@Nonnull` (or its nickname) and `@TypeQualifierDefault(...)` 
+with one or more `ElementType` values:
+* `ElementType.METHOD` for return types of methods;
+* `ElementType.PARAMETER` for value parameters;
+* `ElementType.FIELD` for fields; and
+* `ElementType.TYPE_USE` for any type including type arguments, upper bounds of type parameters and wildcard types.
+
+The default nullability is used when a type itself is not annotated by a nullability annotation, and the default is
+determined by the innermost enclosing element annotated with a type qualifier default annotation with the 
+`ElementType` corresponding to the type usage.
+
+```java
+@Nonnull
+@TypeQualifierDefault({ElementType.METHOD, ElementType.PARAMETER})
+public @interface NonNullApi {
+}
+
+@Nonnull(when = When.MAYBE)
+@TypeQualifierDefault({ElementType.METHOD, ElementType.PARAMETER})
+public @interface NullableApi {
+}
+
+@NullableApi
+interface A {
+    String foo(String x); // fun foo(x: String?): String?
```

Comment:
Probably it's worth adding an example with type arguments (that should rename platform) in case of using `NullableApi`

Reply:
And with generic type arguments, varargs and array elements when they will be supported since it is not obvious to know what types should be annotated.